src/App.tsx

```diff
@@ -1,17 +1,143 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
 
+import { getTodos, addTodo, deleteTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodosList } from './components/TodosList';
+import { FilterBy } from './types/FilterBy';
+import { Footer } from './components/Footer';
+import { TodoAdd } from './components/TodoAdd';
+import { TodoAddCard } from './components/TodoAddCard';
+
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  // Add new Todo
+  const [todosList, setTodoList] = useState<Todo[]>([]);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+
+  // Errors
+  const [titleError, setTitleError] = useState(false);
+  const [addingError, setAddingError] = useState(false);
+  const [deleteError, setDeleteError] = useState(false);
+  const [updateError, setUpdateError] = useState(false);
```

Comment:
instead of 4 errors state, you can create enum with error type and store object with type (and message if needs)